src/mqtt/qmqtt_client_p.cpp

```diff
@@ -429,8 +437,11 @@ void QMQTT::ClientPrivate::onNetworkReceived(const QMQTT::Frame& frm)
 void QMQTT::ClientPrivate::handleConnack(const quint8 ack)
 {
     Q_Q(Client);
-    Q_UNUSED(ack);
-    emit q->connected();
+    if (ack==0) {
+        emit q->connected();
+    }
+   emit q->error(_mqttErrorHash.value(MqttError::MqttError (ack), UnknownError));
```

Comment:
switch-case loop would be *a lot* cheaper

Reply:
emits error() right after connected(). did you ever test your change-set?

Reply:
According to me the `if` needs to be changed to something likes this:
```
if (ack==0)
  emit q->connected();
else
  emit q->error(_mqttErrorHash.value(MqttError::MqttError (ack), UnknownError));
```

Reply:
@mwallnoefer You're right. We don't want to emit the error signal if there is no error.

Reply:
@KonstantinRitt A switch-case would be cheaper indeed. But I guess @yinbaoqiang used this approach, because it was already used for the socket errors. I would accept this pull request first and create the switch-case later.